pkg/build/types/parser/deffile.go

```diff
@@ -131,51 +131,101 @@ func getSectionName(line string) string {
 	return lineSplit[0]
 }
 
-// parseTokenSection splits the token into maximum 2 strings separated by a newline,
-// and then inserts the section into the sections map
-//
-func parseTokenSection(tok string, sections map[string]string) error {
+// parseTokenSection into appropriate components to be placed into a types.Script struct
+func parseTokenSection(tok string, sections map[string]*types.Script, files *[]types.Files) error {
 	split := strings.SplitN(tok, "\n", 2)
 	if len(split) != 2 {
 		return fmt.Errorf("Section %v: Could not be split into section name and body", split[0])
 	}
 
 	key := getSectionName(split[0])
+
+	// parse files differently to allow multiple files sections
+	if key == "files" {
+		f := types.Files{}
+		sectionSplit := strings.SplitN(strings.TrimLeft(split[0], "%"), " ", 2)
+		if len(sectionSplit) == 2 {
+			f.Args = sectionSplit[1]
+		}
+
+		// Files are parsed as a map[string]string
+		filesSections := strings.TrimSpace(split[1])
+		subs := strings.Split(filesSections, "\n")
+		for _, line := range subs {
+			if line = strings.TrimSpace(line); line == "" || strings.Index(line, "#") == 0 {
+				continue
+			}
+			var src, dst string
+			lineSubs := strings.SplitN(line, " ", 2)
+			if len(lineSubs) < 2 {
+				src = strings.TrimSpace(lineSubs[0])
+				dst = ""
+			} else {
+				src = strings.TrimSpace(lineSubs[0])
+				dst = strings.TrimSpace(lineSubs[1])
+			}
+			f.Files = append(f.Files, types.FileTransport{Src: src, Dst: dst})
+		}
+
+		// look through existing files and append to them if they already exist
+		for _, ef := range *files {
+			if ef.Args == f.Args {
+				ef.Files = append(ef.Files, f.Files...)
+				return nil
+			}
+		}
+
+		*files = append(*files, f)
+		return nil
+	}
+
 	if appSections[key] {
 		sectionSplit := strings.SplitN(strings.TrimLeft(split[0], "%"), " ", 3)
 		if len(sectionSplit) < 2 {
 			return fmt.Errorf("App Section %v: Could not be split into section name and app name", sectionSplit[0])
```

Comment:
Again, there should be no caps in return error.

```suggestion
 			return fmt.Errorf("app Section %v: could not be split into section name and app name", sectionSplit[0])
```